### ISSUE TITLE
"Always on Top" works intermittently

### ISSUE THREAD

**Issue by iG8R** [hluk/CopyQ#2524]
**Describe the bug**
The title speaks for itself - CopyQ's "Always on Top" function doesn't work consistently.

**To Reproduce**
1. Enable the "Always on Top" function.
2. Try to actively work with CopyQ for a while.
3. See how the "Always on Top" function stops working.

**Expected behavior**
If the "Always on Top" function is enabled, CopyQ window should always behave according to the enabled function.

**Screenshots**

[CopyQ-Aways-on-Top-Video_2023-10-27_144634.webm](https://github.com/hluk/CopyQ/assets/11407417/d03ce466-f63c-4b3b-8406-b5e83e5c5fe1)

**Version, OS and Environment**
 - CopyQ 7.1.0
 - OS: Microsoft Windows [Version 10.0.19045.2673] 64bit

**Comment by GFDGIT** [comment]
you also need to disable `close_on_unfocus` and `activate_closes` at  the same time.

Or, use this script:
[toggle always_on_top](https://github.com/hluk/CopyQ/issues/2433#issuecomment-1689215068)

**Comment by iG8R** [comment]
Thanks a lot for the advice and script!
But, IMHO, such behavior of CopyQ is somehow contradictory in terms of logic and operation:
 - logic. I think, it's quit weird when the user needs to adjust other parameters for the one to work properly. Shouldn't the application consider all settings and, if there are dependencies, change them accordingly or prioritize the more obvious setting while ignoring the related ones?
- operation. As you can see, I didn't change any other parameters only "Always on Top", but sometimes it works and sometimes it doesn't. If other parameters also affect the behavior of the application, why do they sometimes make changes to the operation of the application and sometimes not?

**Comment by iG8R** [comment]
@GFDGIT 
Can you please give me a hint on how to make it so that after disabling the "Always on Top" function using your script, the main window does not close but remains in the background, visible on the taskbar and through Alt+Tab?

**Comment by iG8R** [comment]
Unfortunately, the script works the same way as when "Always on Top" is manually enabled - sometimes the main window stays on top, sometimes it goes to the background, despite the "Always on Top" option being enabled.

**Comment by iG8R** [comment]
> @GFDGIT Can you please give me a hint on how to make it so that after disabling the "Always on Top" function using your script, the main window does not close but remains in the background, visible on the taskbar and through Alt+Tab?

I figured it out myself - I commented out the following lines in the script and left these parameters disabled in Preferences.
```
//config("activate_closes",on)
//config("close_on_unfocus",on)
```

**Comment by GFDGIT** [comment]
> * logic. I think, it's quit weird when the user needs to adjust other parameters for the one to work properly. 

Sorry, I misunderstood your question.In the video, you enabled `always_ on_ top`, and `close when unfocus` is closed. In this case, even if the focus is lost, the window should not disappear.

> * operation. As you can see, I didn't change any other parameters only "Always on Top", but sometimes it works and sometimes it doesn't. 

I am unable to reproduce the above behavior, may this be a bug? You may want to check if the software has been updated.

**Comment by GFDGIT** [comment]
Besides, let me explain the impact of the other two options on the `always_ on_ top` function:
Note that their priority is also arranged in the following order!

1. `Close mian window` after item activated (in preferences - history, as same as "activate_closes" in the script)
When this option enable, window always close after one item activated, even `always_ on_ top` is enable.

2. `Close when unfocused` (in preferences - general, as same as "close_on_unfocus" in the scrupt)
when this option enable, window always close when lose focus.

3. `always_ on_ top` only controls whether the window is above other windows, without controlling whether it is closed.

In my script, when the  `always_ on_ top` function is enabled, the other two options are turned off simultaneously, achieving a universal sense of  `always_ on_ top`.

**Comment by iG8R** [comment]
> I am unable to reproduce the above behavior, may this be a bug? You may want to check if the software has been updated.

Yes, I think, it's a bug, that is why I've committed this issue.

> Besides, let me explain the impact of the other two options on the `always_ on_ top` function

Thanks a lot!

**Comment by hluk** [comment]
This is possibly a bug in Qt framework. There is already one workaround in CopyQ to keep the window open when the "Always on Top" flag changes.
